backend/database_test.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"github.com/go-redis/redismock/v8"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestConnect(t *testing.T) {
+	t.Run("should check that connection is already established and return error", func(t *testing.T) {
+		//given
+		client, _ := redismock.NewClientMock()
+		d := Database{address: "localhost:8081", password: "", connection: client, ctx: context.Background()}
+
+		//when
+		err := d.Connect()
+
+		//then
+		assert.Equal(t, nil, err)
+	})
+	t.Run("should return error when trying to Ping database", func(t *testing.T) { //todo:test refactor
+
+		//given
+		d := Database{address: "localhost:8081", password: "", ctx: context.Background()}
+
+		//when
+		err := d.Connect()
+
+		//then
+		assert.Error(t, err)
+		assert.NotEqual(t, nil, d.connection)
+	})
+}
+
+func TestInsertToDB(t *testing.T) {
+	t.Run("should return error when connection is not initialized", func(t *testing.T) {
```

Comment:
You used `established` before, so it can be used here in place of `initialised` for consistency